app/javascript/src/display_app_metric.js

```diff
@@ -1,120 +1,140 @@
-    const hour = parseInt(parts[1])
-    const count = counts[key]
+    const parts = key.split(" ");
+    const day = parts[0];
+    const hour = parseInt(parts[1]);
+    const count = counts[key];
 
     dataset.push({
       x: hour,
       y: days.indexOf(day),
       r: Math.sqrt(count) * 2,
-      count
-    })
+      count,
+    });
   }
 
-  return dataset
+  return dataset;
 }
 
-function createChart (dataset) {
-  const ctx = document.getElementById('myChart').getContext('2d')
+function createChart(dataset) {
+  const ctx = document.getElementById("myChart").getContext("2d");
   // eslint-disable-next-line no-unused-vars
   const myChart = new Chart(ctx, {
-    type: 'bubble',
+    type: "bubble",
     data: {
-      datasets: [{
-        label: 'Case Contacts Creation Times in Last Week',
-        data: dataset,
-        backgroundColor: 'rgba(255, 99, 132, 0.2)',
-        borderColor: 'rgba(255, 99, 132, 1)'
-      }]
+      datasets: [
+        {
+          label: "Case Contacts Creation Times in Last Week",
+          data: dataset,
+          backgroundColor: "rgba(255, 99, 132, 0.2)",
+          borderColor: "rgba(255, 99, 132, 1)",
+        },
+      ],
     },
-    options: getChartOptions()
-  })
+    options: getChartOptions(),
+  });
 }
 
-function getChartOptions () {
+function getChartOptions() {
   return {
     scales: {
       x: getXScale(),
-      y: getYScale()
+      y: getYScale(),
     },
     plugins: {
       tooltip: {
         callbacks: {
-          label: getTooltipLabelCallback()
-        }
-      }
-    }
-  }
+          label: getTooltipLabelCallback(),
+        },
+      },
+    },
+  };
 }
 
-function getXScale () {
+function getXScale() {
   return {
     ticks: {
       beginAtZero: true,
-      stepSize: 1
-    }
-  }
+      stepSize: 1,
+    },
+  };
 }
 
-function getYScale () {
+function getYScale() {
   return {
     ticks: {
       beginAtZero: true,
       stepSize: 1,
-      callback: getYTickCallback()
-    }
-  }
+      callback: getYTickCallback(),
+    },
+  };
 }
 
-function getYTickCallback () {
+function getYTickCallback() {
   return function (value, index, values) {
-    return days[value]
-  }
+    return days[value];
+  };
 }
 
-function getTooltipLabelCallback () {
+function getTooltipLabelCallback() {
   return function (context) {
-    const datum = context.dataset.data[context.dataIndex]
-    return datum.count + ' case contacts created on ' + days[datum.y] + ' at ' + datum.x + ':00'
+    const datum = context.dataset.data[context.dataIndex];
+    return datum.count + " case contacts created on " + days[datum.y] + " at " + datum.x + ":00";
+  };
+}
+
+function getErrorMessage(xhr) {
+  // Get the specific error message based on status codes or response from server
+  if (xhr.status === 404) {
+    return "The requested resource was not found";
+  } else if (xhr.status === 500) {
+    return "Internal server error occurred.";
+  } else {
+    return "An unknown error occurred.";
```

Comment:
Could you display an error that states the chart failed to load always and print these specific errors in the console instead?